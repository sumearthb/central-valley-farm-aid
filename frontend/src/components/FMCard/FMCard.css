.fm-cards-container {
    display: grid;
    grid-template-columns: repeat(3, minmax(0, 1fr)); /* Creates a grid with 3 columns of equal width */
    gap: 20px; /* Adjust the gap between cards */
  }
  
  /* Class that applies to each card */
  .fm-card {
    border: 1px solid #ccc;
    padding: 8px;
    margin: 8px;
    text-align: start;
    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
    background-color: #fff;
    transition: transform 0.2s ease-in-out;
    width: 300px;
    height: 500px;
  }
  
  /* Affects image */
  .fm-card img {
    display: block;
    width: 30vh;
    height: 30vh;
    border: 1px solid #ddd;
    border-radius: 4px;
    margin-bottom: 12px;
  }
  
  /* Affects font size of title of card*/ 
  .fm-card h2 {
    font-size: 1.5rem;
    margin: 0;
    color: #333;
  }

  /* Class that defines container for centering child elements */
  .centered-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 80vh; /* Sets specific height for centering vertically */
  }