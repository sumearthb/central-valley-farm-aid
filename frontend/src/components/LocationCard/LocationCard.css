.location-cards-container {
  display: grid;
  grid-template-columns: repeat(3, minmax(0, 1fr)); /* Creates a grid with 3 columns of equal width */
  gap: 20px; /* Adjust the gap between cards */
}

/* Class that applies to each card */
.location-card {
  border: 1px solid #ccc;
  margin: 1rem;
  text-align: start;
  box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
  background-color: #fff;
  transition: transform 0.2s ease-in-out;
  width: 21rem;
}

/* Affects image */
.location-card img {
  display: block;
  width: 100%;
  height: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
  align-self: center;
}

/* Affects font size of title of card*/ 
.location-card h2 {
  font-size: 1.5rem;
  margin: 0;
  color: #333;
}

/* Class that defines container for centering child elements */
.centered-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 80vh; /* Sets specific height for centering vertically */
}

.location-card-body {
  text-align: left;
  background-color: #F0FDEE;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}