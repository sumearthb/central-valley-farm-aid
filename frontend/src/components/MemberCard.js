import PropTypes from "prop-types";
import React from "react";
import Card from "react-bootstrap/Card";
import ListGroup from 'react-bootstrap/ListGroup';

function MemberCard(props) {
  const {
    name,
    username,
    bio,
    commits,
    issues,
  } = props;

  return (
    <Card className="member-card">
      <Card.Img className="member-card-image" variant="top"
        src={`${process.env.PUBLIC_URL}/member-pics/${username}.jpg`} />
      <Card.Body className="member-card-body">
        <Card.Title className="member-card-name">
          {name}
        </Card.Title>
        <Card.Text className="member-card-gitlab">
          <a href={"https://gitlab.com/" + username}>
            {username}
          </a>
        </Card.Text>
        <Card.Text className="member-card-bio">
          {bio}
        </Card.Text>
        <ListGroup horizontal id="member-card-stats">
          <ListGroup.Item>{`${commits} Commits`}</ListGroup.Item>
          <ListGroup.Item>{`${issues} Issues`}</ListGroup.Item>
        </ListGroup>
      </Card.Body>
    </Card>
  );
}

MemberCard.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  commits: PropTypes.number.isRequired,
  issues: PropTypes.number.isRequired,
};

export default MemberCard;
