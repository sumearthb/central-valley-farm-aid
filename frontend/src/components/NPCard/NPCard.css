.np-cards-container {
    display: grid;
    grid-template-columns: repeat(3, minmax(0, 1fr)); /* Creates a grid with 3 columns of equal width */
    gap: 20px; /* Adjust the gap between cards */
  }
  
  /* Class that applies to each card */
  .np-card {
    border: 1px solid #ccc;
    margin: 1rem;
    text-align: start;
    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2);
    background-color: #fff;
    transition: transform 0.2s ease-in-out;
    width: 21rem;
  }

  /* Affects image */
  .np-card img {
    display: block;
    width: 100%;
    height: auto;
    border: 1px solid #ddd;
    border-radius: 4px;
    align-self: center;
  }
  
  /* Affects font size of title of card*/ 
  .np-card h2 {
    font-size: 1.5rem;
    margin: 0;
    color: #333;
  }

  .np-card h3 {
    font-size: 1.5rem;
    margin: 0;
    color: #333;
    font-size: .8em;
    margin-bottom: 13.5px;
  }

  /* Class that defines container for centering child elements */
  .centered-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 80vh; /* Sets specific height for centering vertically */
  }

  .np-card-details {
    text-align: left;
  }