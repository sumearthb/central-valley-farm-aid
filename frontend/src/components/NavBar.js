import "../styles/NavBar.css";
import React from "react";
import { Link } from "react-router-dom";
import {Container, Nav, Navbar, Form, Button} from "react-bootstrap";

function NavBar() {
  return (
    <Navbar expand="lg" className="navbar">
      <Container>
        <Navbar.Brand>
          <Link className="navbar-link"
            to="/">
            Central Valley Farm Aid
          </Link>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Link href="/">
              Home
            </Nav.Link>
            <Nav.Link href="/about">
              About
            </Nav.Link>
            <Nav.Link href="/locations">
              Locations
            </Nav.Link>
            <Nav.Link href="/nonprofits">
              Nonprofits
            </Nav.Link>
            <Nav.Link href="/farmersmarkets">
              Farmers' Markets
            </Nav.Link>
          </Nav>
          <Form className="d-flex justify-content-end">
            <Form.Control type="search" id="searchText" placeholder="Search..." 
                className="mx-2" aria-label="Search"></Form.Control>
            <Button type="submit" variant="dark">Search</Button>
          </Form>

        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default NavBar;
