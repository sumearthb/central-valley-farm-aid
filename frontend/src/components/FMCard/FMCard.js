import React from "react";
import "./FMCard.css";
import { Card, Button } from "react-bootstrap";
import { Link } from "react-router-dom";

const FMCard = ({ market }) => {
  return (
    <Card className="fm-card">
      <Card.Img className="fm-card-image" src={`https://maps.googleapis.com/maps/api/place/photo?maxwidth=1000&photo_reference=${market.photo_references[0]}&key=AIzaSyBMJJbFxLfnX8DpE_BGF2dF8t5aWSQJOOs`}/>
      <Card.Body className="fm-card-body">
        <Card.Title className="fm-card-title">
          {market.listing_name}
        </Card.Title>
        <Card.Text className="fm-card-rating">
          Rating: {market.rating}
        </Card.Text>
        <Card.Text className="fm-card-indoor">
          Indoor Status: {market.location_indoor}
        </Card.Text>
        <Card.Text className="fm-card-production">
          Crop Production: {market.specialproductionmethods}
        </Card.Text>
        <Card.Text className="fm-card-fnap">
          FNAP Status: {market.fnap.replace("0", "None")}
        </Card.Text>
        <Card.Text className="fm-card-wheelchair">
          Wheelchair Accessible: {market.wheelchair_accessible}
        </Card.Text>
        <Link
          to={`/farmersmarkets/${market.name}`}
          style={{ textDecoration: "none" }}>
          <Button variant="primary" style={{ backgroundColor: '#69A765'}} className="align-self-end border-white">More info</Button>
        </Link>
      </Card.Body>
    </Card>
  );
};
  
export default FMCard;