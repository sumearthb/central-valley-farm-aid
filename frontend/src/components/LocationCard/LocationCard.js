import React from "react";
import "./LocationCard.css";
import { Card, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const LocationCard = ({ location }) => {
  return (
    <Card className="location-card">
      <Card.Img className="location-card-image" src={location.map.replace("100px", "400px")}/>
      <Card.Body className="location-card-body">
        <Card.Title className="location-card-title">
          {location.name + " County"}
        </Card.Title>
        <Card.Text className="location-card-countyseat">
          County Seat: {location.county_seat}
        </Card.Text>
        <Card.Text className="location-card-crop">
          # of Crops: {location.crops.crops.length}
        </Card.Text>
        <Card.Text className="location-card-population">
          Population: {location.population}
        </Card.Text>
        <Card.Text className="location-card-established">
          Established: {location.est}
        </Card.Text>
        <Card.Text className="location-card-area">
          Area: {location.area} sq mi
        </Card.Text>
        <Link
          to={`/locations/${location.name}`}
          style={{ textDecoration: "none" }}
        >
          <Button variant="primary" style={{ backgroundColor: '#69A765'}} className="align-self-end border-white">More info</Button>
        </Link>
        {/* Add more details if needed */}
      </Card.Body>
    </Card>
  );
};
  
export default LocationCard;