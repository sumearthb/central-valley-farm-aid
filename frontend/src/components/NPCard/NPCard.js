import React from "react";
import "./NPCard.css";
import { Card, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const NPCard = ({ nonprofit }) => {
  return (
    // specifies CSS class used
    <Card className="np-card">
      <Card.Img className="np-card-image" src={`https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photo_reference=${nonprofit.photo_references.photos[0]}&key=AIzaSyBMJJbFxLfnX8DpE_BGF2dF8t5aWSQJOOs`}/>
      <Card.Body className="np-card-body">
        <Card.Title className="np-card-name">
          {nonprofit.charityName}
        </Card.Title>
        <Card.Text className="np-card-category">
          Category: {nonprofit.category}
        </Card.Text>
        <Card.Text className="np-card-city">
          City: {nonprofit.city}
        </Card.Text>
        <Card.Text className="np-card-zip">
          Zip Code: {nonprofit.zipCode}
        </Card.Text>
        <Card.Text className="np-card-ein">
          Employee Identification Number: {nonprofit.ein}
        </Card.Text>
        <Link
          to={nonprofit.url}
          style={{ textDecoration: "none" }}>
          {nonprofit.url}
        </Link>
        <Link
          to={`/nonprofits/${nonprofit.charityName}`}
          style={{ textDecoration: "none" }}>
          <Button variant="primary" style={{ backgroundColor: '#69A765'}} className="align-self-end border-white">More info</Button>
        </Link>
      </Card.Body>
    </Card>
  );
};

export default NPCard;