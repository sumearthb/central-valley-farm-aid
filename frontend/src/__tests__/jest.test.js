/**
 * @jest-environment jsdom
 */

import React from "react";

import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import {BrowserRouter as Router, Route, Routes} from "react-router-dom"
import NavBar from "../components/NavBar.js"
import LocationCard from "../components/LocationCard/LocationCard.js";
import NPCard from "../components/NPCard/NPCard.js";
import FMCard from "../components/FMCard/FMCard.js";
import MemberCard from "../components/MemberCard.js";
import test_location from "./location.json";
import test_nonprofit from "./nonprofit.json";
import test_farmersmarket from "./farmersmarket.json";
import test_member from "./test_member.json";

test("test location card snapshot", () => {
  const location = test_location;
  const tree = renderer
    .create(
      <Router>
        <LocationCard
        location={location} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
  
test("test nonprofit card snapshot", () => {
  const nonprofit = test_nonprofit;
  const tree = renderer
    .create(
      <Router>
        <NPCard nonprofit={nonprofit} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
  
test("test farmers market card snapshot", () => {
  const market = test_farmersmarket;
  const tree = renderer
    .create(
      <Router>
        <FMCard market={market} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
  
test("test member card snapshot", () => {
  const member = test_member;
  const tree = renderer
    .create(
      <Router>
        <MemberCard member={member} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Test whether the Navbar works", () => {
    const tree = renderer.create(
      <Router>
        <NavBar />
      </Router>
    );
    expect(tree).toMatchSnapshot();
});
  
  test("Test whether the Navbar contains links to About, Home, Parks, Animals, and States", async () => {
    render(
      <Router>
        <NavBar />
      </Router>
    );
    const navbar = screen.getByTestId("navbar");
    expect(navbar).toHaveTextContent("Home");
    expect(navbar).toHaveTextContent("About");
    expect(navbar).toHaveTextContent("Locations");
    expect(navbar).toHaveTextContent("Nonprofits");
    expect(navbar).toHaveTextContent("Farmers' Markets");
});
  

