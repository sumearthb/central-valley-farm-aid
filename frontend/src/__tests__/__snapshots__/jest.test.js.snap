// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test whether the Navbar works 1`] = `
<nav
  className="navbar navbar navbar-expand-lg navbar-light"
  data-testid="navbar"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <a
        className="navbar-link"
        href="/"
        onClick={[Function]}
      >
        Central Valley Farm Aid
      </a>
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/about"
          disabled={false}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/locations"
          disabled={false}
          href="/locations"
          onClick={[Function]}
        >
          Locations
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/nonprofits"
          disabled={false}
          href="/nonprofits"
          onClick={[Function]}
        >
          Nonprofits
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/farmersmarkets"
          disabled={false}
          href="/farmersmarkets"
          onClick={[Function]}
        >
          Farmers' Markets
        </a>
      </div>
      <form
        className="d-flex justify-content-end"
      >
        <input
          aria-label="Search"
          className="mx-2 form-control"
          id="searchText"
          placeholder="Search..."
          type="search"
        />
        <button
          className="btn btn-dark"
          disabled={false}
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</nav>
`;

exports[`test farmers market card snapshot 1`] = `
<div
  className="fm-card card"
>
  <img
    className="card-img fm-card-image"
    src="https://maps.googleapis.com/maps/api/place/photo?maxwidth=1000&photo_reference=AcJnMuGiiYANDybm3GbGbxaGeqtXp94WoDNUXBamATFrNeBWsZ4zauNb2jiFGJk3Oh6xzAxn37UdV9LVFtX6tUM7RaiZWiexTxJ16WYQ4fbTNvHjUD-mclqGR6zJtYAXK1cl4va4rEOPusfW3vRaoH7oNreXQDcvh4_kE9VHIZpTJ4FLrfwo&key=AIzaSyBMJJbFxLfnX8DpE_BGF2dF8t5aWSQJOOs"
  />
  <div
    className="fm-card-body card-body"
  >
    <div
      className="fm-card-title card-title h5"
    >
      Benica Certified Farmers Market
    </div>
    <p
      className="fm-card-rating card-text"
    >
      Rating: 
      4.4
    </p>
    <p
      className="fm-card-indoor card-text"
    >
      Indoor Status: 
      No Indoor;
    </p>
    <p
      className="fm-card-production card-text"
    >
      Crop Production: 
      Organic (USDA Certified);
    </p>
    <p
      className="fm-card-fnap card-text"
    >
      FNAP Status: 
      WIC;SNAP;Accept EBT at a central location;;
    </p>
    <p
      className="fm-card-wheelchair card-text"
    >
      Wheelchair Accessible: 
      True
    </p>
    <a
      href="/farmersmarkets/undefined"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className="align-self-end border-white btn btn-primary"
        disabled={false}
        style={
          Object {
            "backgroundColor": "#69A765",
          }
        }
        type="button"
      >
        More info
      </button>
    </a>
  </div>
</div>
`;

exports[`test location card snapshot 1`] = `
<div
  className="location-card card"
>
  <img
    className="card-img location-card-image"
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/Map_of_California_highlighting_Alameda_County.svg/400px-Map_of_California_highlighting_Alameda_County.svg.png"
  />
  <div
    className="location-card-body card-body"
  >
    <div
      className="location-card-title card-title h5"
    >
      Alameda County
    </div>
    <p
      className="location-card-countyseat card-text"
    >
      County Seat: 
      Oakland
    </p>
    <p
      className="location-card-crop card-text"
    >
      # of Crops: 
      8
    </p>
    <p
      className="location-card-population card-text"
    >
      Population: 
      1628997
    </p>
    <p
      className="location-card-established card-text"
    >
      Established: 
      1853
    </p>
    <p
      className="location-card-area card-text"
    >
      Area: 
      738
       sq mi
    </p>
    <a
      href="/locations/Alameda"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className="align-self-end border-white btn btn-primary"
        disabled={false}
        style={
          Object {
            "backgroundColor": "#69A765",
          }
        }
        type="button"
      >
        More info
      </button>
    </a>
  </div>
</div>
`;

exports[`test member card snapshot 1`] = `
<div
  className="member-card card"
>
  <img
    className="card-img-top member-card-image"
    src="undefined/member-pics/martinmurtiono.jpg"
  />
  <div
    className="member-card-body card-body"
  >
    <div
      className="member-card-name card-title h5"
    >
      Martin Murtiono
    </div>
    <p
      className="member-card-gitlab card-text"
    >
      <a
        href="https://gitlab.com/martinmurtiono"
      >
        martinmurtiono
      </a>
    </p>
    <p
      className="member-card-role card-text"
    >
      Full Stack
    </p>
    <p
      className="member-card-bio card-text"
    >
      bio
    </p>
    <div
      className="list-group list-group-horizontal"
      id="member-card-stats"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        123 Commits
      </div>
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        123 Issues
      </div>
      <div
        className="list-group-item"
        data-rr-ui-event-key={null}
        onClick={[Function]}
      >
        123 Unit Tests
      </div>
    </div>
  </div>
</div>
`;

exports[`test nonprofit card snapshot 1`] = `
<div
  className="np-card card"
>
  <img
    className="card-img np-card-image"
    src="https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photo_reference=AcJnMuEdoMCokENWhhAG_lPj8GznGzxvezJqdyvJdaQAP36o3cZlH8qKqvu45jAQBM2YO4SlYPaITSvvngRKxYEkDhws91_fyIJgXEMnWH6PXTYYPLS3BqeHt9_flA5_Eqd9Yq39JckzTHkbEC-4B6DFAkOvbnJ4C90ErcS0qcVSp4JOUmi2&key=AIzaSyBMJJbFxLfnX8DpE_BGF2dF8t5aWSQJOOs"
  />
  <div
    className="np-card-body card-body"
  >
    <div
      className="np-card-name card-title h5"
    >
      R & S FARMERS MARKET R & S GILROY FARMERS MARKET
    </div>
    <p
      className="np-card-category card-text"
    >
      Category: 
      Food, Agriculture and Nutrition
    </p>
    <p
      className="np-card-city card-text"
    >
      City: 
      SALINAS
    </p>
    <p
      className="np-card-zip card-text"
    >
      Zip Code: 
      93907-9016
    </p>
    <p
      className="np-card-ein card-text"
    >
      Employee Identification Number: 
      881734546
    </p>
    <a
      href="http://www.orghunter.com/organization/881734546"
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      http://www.orghunter.com/organization/881734546
    </a>
    <a
      href="/nonprofits/R & S FARMERS MARKET R & S GILROY FARMERS MARKET"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className="align-self-end border-white btn btn-primary"
        disabled={false}
        style={
          Object {
            "backgroundColor": "#69A765",
          }
        }
        type="button"
      >
        More info
      </button>
    </a>
  </div>
</div>
`;
